grakn-engine/src/main/java/ai/grakn/engine/GraknCreator.java

```diff
@@ -119,7 +121,21 @@ public synchronized GraknEngineServer instantiateGraknEngineServer(Runtime runti
             EngineGraknTxFactory factory = instantiateGraknTxFactory(graknEngineConfig, lockProvider);
             PostProcessor postProcessor = postProcessor(metricRegistry, graknEngineConfig, factory, jedisPool, lockProvider);
             TaskManager taskManager = instantiateTaskManager(metricRegistry, graknEngineConfig, engineID, factory, jedisPool, postProcessor);
-            HttpHandler httpHandler = new HttpHandler(graknEngineConfig, sparkService, factory, metricRegistry, graknEngineStatus, taskManager, postProcessor);
+
+            int grpcPortPort = graknEngineConfig.getProperty(GraknConfigKey.GRPC_PORT);
+            GrpcServer grpcServer;
+            try {
+                grpcServer = GrpcServer.create(grpcPortPort, engineGraknTxFactory);
+            } catch (IOException e) {
+                // Thrown if unable to bind to the given port
+                throw new IllegalStateException(e);
```

Comment:
Just out of curiousity: why is the IllegalState exception better than the IO exception here?

Reply:
I said this in person, but I just wanted an unchecked exception. The `IOException` is probably more correct though.